### DISCUSSION TITLE
Detect CMYK vs RGB Color Space for JPEG and TIFF?

### DISCUSSION THREAD

**Discussion by pha3z** [SixLabors/ImageSharp#2278] [Q&A]
It seems like Image.Identify(filepath) should allow me to detect whether a JPEG is CMYK or RGB, but I can't seem to figure it out.

How do I detect CMYK vs RGB of a JPEG?

**Comment by pha3z** [comment]
I also need to detect rgb vs cmyk TIFF

**Reply by JimBobSquarePants** [reply]
Getting per-frame Tiff metadata is currently unavailable without decoding. If you can raise an issue so we can cover that I'd appreciate it.

**Reply by pha3z** [reply]
Looks like there's no way to raise a Feature Request Issue.  The button directs to the discussions board.  Should I be posting this as an idea???  Or perhaps it should be raised as a bug??

**Reply by JimBobSquarePants** [reply]
Sorry yeah, open a discussion in ideas and put as much detail as you can in your expectations. Thanks!

**Comment by JimBobSquarePants** [comment]
You're looking for [`JpegMetadata.ColorType`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Formats.Jpeg.JpegMetadata.html#SixLabors_ImageSharp_Formats_Jpeg_JpegMetadata_ColorType)

```c#
IImageInfo info = Image.Identify(filepath);
JpegMetadata meta = info.Metadata.GetJpegMetadata();
JpegColorType? colorType = meta.ColorType;
```
